testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_transactions.py

```diff
@@ -0,0 +1,420 @@
+    '''
+
+    xattrs_enabled = params_from_base_test_setup['xattrs_enabled']
+    if not xattrs_enabled:
+        pytest.skip('this test require --xattrs flag')
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    db = params_from_base_test_setup["db"]
+    cluster_helper = ClusterKeywords(cluster_config)
+    topology = cluster_helper.get_cluster_topology(cluster_config)
+    cluster_servers = topology["couchbase_servers"]
+    cbs_one_url = cluster_servers[0]
+    cb_server = couchbaseserver.CouchbaseServer(cbs_one_url)
+
+    sg_db = "db"
+    username = "autotest"
+    password = "password"
+    channels = ["TRANSACTIONS"]
+    num_of_docs = 4
+    doc_ids = ""
+
+    sg_client = MobileRestClient()
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+    bucket = cb_server.get_bucket_names()[0]
+
+    # disable replicas and rebalance the server
+    cb_server.disable_replicas(bucket)
+    cb_server.rebalance_server(cluster_servers)
+
+    # randomly  generate replication ids
+    for i in range(0, num_of_docs):
+        doc_id = "txn_id_" + str(time.time())
+        doc_ids = doc_id + "," + doc_ids
+    opt_doc_ids = doc_ids.replace("txn_id_", "insert-txn_id_")
+    removable_id = doc_ids.split(",")[1]
+    # 1: Insert a series of documents (A, B, C) in a transaction . CBL is not syncing as yet. SGW does import
+    # CBL is not syncing as yet. SGW does import
+    transactions_app_dir = "resources/data/transaction-performer.jar"
+    doc_body = document.create_doc(doc_id=doc_id, content="doc1", channels=channels)
+    data = json.dumps(doc_body)
+    cmd = "java -cp {} com.couchbase.transaction clusterip={} operationdocids={} bucket={} doccontent='{}'".format(transactions_app_dir, c.servers[0].host, opt_doc_ids, bucket, data)
+    subprocess.check_output(cmd, shell=True)
+
+    # 2: Update (edits) a series of documents in this order (A->A',B-> B' (tombstoned) ,C-> C') in a transaction At the exact same time, CBL starts pulling docs via SGW.
+    opt_doc_ids = doc_ids.replace("txn_id_", "replace-txn_id_")
+    opt_doc_ids = opt_doc_ids.replace("replace-{}".format(removable_id), "remove-{}".format(removable_id))
+    cmd = "java -cp {} com.couchbase.transaction clusterip={} operationdocids={} bucket={}".format(transactions_app_dir, c.servers[0].host, opt_doc_ids, bucket)
+    with ThreadPoolExecutor(max_workers=2) as tpe:
+        shell_cmd_task = tpe.submit(shell_process, cmd)
+
+        # 4. Replicate from SGW to CBL
```

Comment:
is this step 3? seems the comments are not matching the steps listed in summary?

Reply:
updated comment